### ISSUE TITLE
Torchscript super-resolution model  inference not working

### ISSUE THREAD

**Issue by voidpunk** [LaurentMazare/tch-rs#607]
I'm trying to use tch-rs to do inference on a model I trained in Python and successfully used for inference with the C++ API after compiling it with Torchscript. However, when doing the exact same steps on the exact same traced model with tch-rust I obtain a quite strange looking results. Here's the comparison:

 ## C++ API
![srcpp](https://user-images.githubusercontent.com/81917075/211778658-6ef03428-42fc-45ea-ba43-fc75deda6560.png)

## tch-rs API
![srrust](https://user-images.githubusercontent.com/81917075/211778679-5ae648dc-89f6-4ab4-b7fd-34fa061c940a.png)

## model
Example: model rewritten with sequential to make sure it was as straightforward as possible:

```python
import torch
from torch import nn

model = nn.Sequential(
        nn.Conv2d(in_channels=1, out_channels=64, kernel_size=5, padding=2),
        nn.PReLU(num_parameters=64),
        nn.Conv2d(in_channels=64, out_channels=52, kernel_size=3, padding=1),
        nn.PReLU(num_parameters=52),
        nn.Conv2d(in_channels=52, out_channels=40, kernel_size=3, padding=1),
        nn.PReLU(num_parameters=40),
        nn.Conv2d(in_channels=40, out_channels=28, kernel_size=3, padding=1),
        nn.PReLU(num_parameters=28),
        nn.Conv2d(in_channels=28, out_channels=16, kernel_size=3, padding=1),
        nn.PixelShuffle(upscale_factor=4)
    )

model.load_state_dict(torch.load("./model.pth"))
model.eval()
example = torch.rand(1, 1, 1000, 1000)
traced_script_module = torch.jit.trace(model, example)
traced_script_module.save("traced_model.pth")
```

### Rust
Example: inference relevant Rust lines

```rust
let tensor = tch::Tensor::of_slice(v32in.as_slice());
let shape: Vec<i64> = vec![1, 1, h, w];
let input = tensor.reshape(&shape);
let model = tch::CModule::load("/traced_model.pth").unwrap();
let output = input.apply(&model).squeeze();
```

For tch-rust I tried inference with both `let output = model.forward_ts(&[input]).unwrap().squeeze();` and `let output = input.apply(&model).squeeze();`, with the same result.

### C++
Example: inference relevant C++ lines

```cpp
at::Tensor x_tensor = torch::from_blob(f32img.data, {1, 1, height, width}, torch::kFloat);
std::vector<torch::jit::IValue> inputs;
inputs.push_back(x_tensor);
at::Tensor outputs = model.forward(inputs).toTensor().squeeze();
```

In both cases the images are single channel 16 bit images which I convert to f32 and normalize. In both cases I make sure that I can reconstruct the same image if don't pass it to the model, so the issue isn't in the pre or post processing steps. The traced model file I use is the same, as well as the input image of course. I also checked that the input, output, as well as the weights are effectively f32, and they are.

I ran the [examples](https://github.com/LaurentMazare/tch-rs/tree/main/examples/jit) without issues, so the problem must be on the way this architecture is loaded I guess (maybe the issue could be the pixelshuffle layer?).

## infos

- OS: Ubuntu (WSL)
- rustc: 1.66.1
- tch: 0.10.1
- libtorch: 1.13.0

Thank you 😊

**Comment by LaurentMazare** [comment]
The torchscript bits of `tch` should actually just be thin wrappers around the C++ version so it's indeed suprising to have such discrepancies. It might be that image loading doesn't do exactly the same thing, e.g. one generates pixel fvalues from 0 to 255 and the other from 0 to 1 or something like this (also not sure how well grayscale works in `tch`), I guess you'll probably want to print some of the intermediate tensors to check if things line up well.

**Comment by voidpunk** [comment]
As I said, I convert the input to f32/float and normalize it (so, interval [0,1]), and I check both the type and the intervals in the input on both C++ and tch-rs. The grayscale is just a one-channel image or a 3 channels image where all channels have the same value. As I said, in my case it's a single channel image. The issue must be how the model is loaded or how it run the forward pass, since the values explode there:

C++ input max value:
```cpp
std::cout << "min in: " <<  torch::min(input) << "\n" << "max in: " <<  torch::max(input) << std::endl;
at::Tensor output = model.forward(input).toTensor().squeeze();
std::cout << "min out: " <<  torch::min(output) << "\n" << "max out: " << torch::max(output) << std::endl;
```
output:
```
min in: 0
max in: 0.929091
min out: -0.0858269
max out: 0.960142
```

tch-rs input max value:
```rust
println!("min in: {:?}\nmax in {:?}", tch::Tensor::min(&input), tch::Tensor::max(&input));
let output = input.apply(&model).squeeze();
println!("min out: {:?}\nmax out: {:?}", tch::Tensor::min(&output), tch::Tensor::max(&output));
```
output:
```
min in: 0
max in: 0.92900654296875
min out: -2.4308676719665527
max out: 2.8395004272460938
```

the only difference is the forward line...
